src/plugins/intel_cpu/src/nodes/llm_mlp.cpp

```diff
@@ -0,0 +1,652 @@
+        do_config(nullptr);
+    }
+    void* last_cfg = nullptr;
+    void do_config(void* cfg) {
+        if (cfg != last_cfg) {
+            (*this)(cfg);
+            last_cfg = cfg;
+        }
+    }
+};
+
+template <typename T>
+class Singleton {
+public:
+    static T& get() {
+        static T instance;
+        return instance;
+    }
+};
+
+static PlainTensor& getC(int ithr) {
+    static std::vector<PlainTensor> all_C(parallel_get_max_threads());
+    return all_C[ithr];
+}
+
+struct Work {
+    std::vector<PlainTensor> weights;  // ov::bfloat16 weights for current thread
+
+    std::shared_ptr<std::atomic_int> sync_flag;
+    int n0 = 0;
+    int n1 = 0;
+    int k0 = 0;
+    int k1 = 0;
+    int BN = 0;
+    int blk_K_size = 0;
+    int output_id;
+    ov::bfloat16* p_raw_weights;
+    operator bool() {
+        return BN > 0;
+    }
+
+    MKernel& get_MKernel() {
+        constexpr int BM = 256;
+        static MKernel jit_amx0(BM);
+        return jit_amx0;
+    }
+
+    // input : weight [N, K], setup repacks range of N [n_start, n_end)
+    template <typename T>
+    void setup(T* p_weight, int stride) {
+        auto& mkernel = get_MKernel();
+        auto num_blk_K = (k1 - k0) / blk_K_size;
+        auto* pw = p_weight + n0 * stride / sizeof(T) + k0;
+
+        weights.resize(num_blk_K);
+        for (int k = 0; k < num_blk_K; k++) {
+            mkernel.prepareB(weights[k], pw + k * blk_K_size, stride, BN, blk_K_size);
+        }
+
+        for (int Mtails = 0; Mtails < 32; Mtails++) {
+            mkernel.tile_config_M(m_tcfg[Mtails], Mtails == 0 ? 32 : Mtails);
+        }
+    }
+
+    TileConfig m_tcfg[32];
+
+    void run(int M, uint8_t* pA, int strideA, PlainTensor& C) {
+        auto& mkernel = get_MKernel();
+
+        int num_blk_K = (k1 - k0) / blk_K_size;
+
+        auto Mtails = M % 32;
+        auto Mbody = M - Mtails;
+
+        auto C_M = Mbody + (Mtails ? 32 : 0);
+        C.resize<float>({static_cast<size_t>(C_M), static_cast<size_t>(BN)});
+        auto pC = reinterpret_cast<uint8_t*>(C.ptr_v());
+
+        auto& tile_configer = (Singleton<AutoTileConfiger>::get());
+
+        pA += k0 * sizeof(ov::bfloat16);
+        bool do_accumulation = false;
+
+        for (int ki = 0; ki < num_blk_K; ki++) {
+            PlainTensor& blockB = weights[ki];
+            PlainTensor& blockB1 = weights[(ki + 1) < num_blk_K ? (ki + 1) : ki];
+            if (Mbody) {
+                tile_configer.do_config(&m_tcfg[0]);
+                mkernel.run(Mbody,
+                            pA + ki * blk_K_size * sizeof(ov::bfloat16),
+                            strideA,
+                            blockB,
+                            pC,
+                            C.stride_bytes(0),
+                            reinterpret_cast<uint8_t*>(blockB1.ptr_v()),
+                            do_accumulation);
+            }
+
+            if (Mtails) {
+                tile_configer.do_config(&m_tcfg[Mtails]);
```

Comment:
If multiple threads change here at the same time, maybe the config is not updated correctly? Maybe making the AutoTileConfiger thread local will be better?

Reply:
Good insight! will fix!